cobra/flux_analysis/loopless.py

```diff
@@ -1,6 +1,71 @@
 from ..core import Reaction, Metabolite
 from ..manipulation.modify import convert_to_irreversible
+import numpy as np
 from six import iteritems
+from cobra.util import nullspace, add_to_solver
+from sympy.core.singleton import S
+
+
+def add_loopless(model, zero_cutoff=1e-12):
+    """Add variables and constraints to make a model thermodynamically
+    feasible. This removes flux loops. The used formulation is described
+    in [1]_.
+
+    Parameters
+    ----------
+    model : a cobra model
+        The model to which to add the constraints.
+    zero_cutoff : positive float, optional
+        Cutoff used for null space. Coefficients with an absolute value smaller
+        than `zero_cutoff` are considered to be zero.
+
+    Returns
+    -------
+    Nothing
+
+    References
+    ----------
+    .. [1] Elimination of thermodynamically infeasible loops in steady-state
+       metabolic models.
+       Schellenberger J, Lewis NE, Palsson BØ.
+       Biophys J. 2011 Feb 2;100(3):544-53.
+       doi: 10.1016/j.bpj.2010.12.3707.
+       Erratum in: Biophys J. 2011 Mar 2;100(5):1381.
+    """
+    internal = [i for i, r in enumerate(model.reactions) if not r.boundary]
+    Sint = model.S[:, np.array(internal)]
+    Nint = nullspace(Sint).T
+    max_bound = max(max(abs(b) for b in r.bounds) for r in model.reactions)
+    prob = model.solver.interface
+
+    # Add indicator variables and new constraints
+    to_add = []
+    for i in internal:
+        rxn = model.reactions[i]
+        # indicator variable a_i
+        indicator = prob.Variable("indicator_" + rxn.id, type="binary")
+        # -M*(1 - a_i) <= v_i <= M*a_i
+        on_off_constraint = prob.Constraint(
+            rxn.flux_expression - max_bound * indicator,
+            lb=-max_bound, ub=0, name="on_off_" + rxn.id)
+        # -(max_bound + 1) * a_i + 1 <= G_i <= -(max_bound + 1) * a_i + 1000
+        delta_g = prob.Variable("delta_g_" + rxn.id)
+        delta_g_range = prob.Constraint(
+            delta_g + (max_bound + 1) * indicator,
+            lb=1, ub=max_bound, name="delta_g_range_" + rxn.id)
+        to_add.extend([indicator, on_off_constraint, delta_g, delta_g_range])
+
+    add_to_solver(model, to_add)
+
+    # Add nullspace constraints for G_i
+    for i, row in enumerate(Nint):
+        name = "nullspace_constraint_" + str(i)
+        nullspace_constraint = prob.Constraint(S.Zero, lb=0, ub=0, name=name)
+        add_to_solver(model, [nullspace_constraint])
+        coefs = {model.solver.variables[
+            "delta_g_" + model.reactions[ridx].id]: row[i]
+            for i, ridx in enumerate(internal) if abs(row[i]) > zero_cutoff}
+        model.solver.constraints[name].set_linear_coefficients(coefs)
```

Comment:
Already is due to the add_to_solver above. Had to do it like that because you can't set linear coefficients for a Constraint that is not part of an optlang.Model yet.